### PR TITLE
add isenabled to menuview

### PR DESCRIPTION
**Pull request by noodlemoodle**

### CODE REVIEW

File: src/foam/u2/view/MenuView.js [hunk 1]
```diff
@@ -27,6 +27,27 @@ foam.CLASS({
     {
       name: 'themeIcon',
       expression: function(menu) { return menu.themeIcon || ''; }
+    },
+    {
+      class: 'Function',
+      generateJava: false,
+      name: 'isEnabled',
+      label: 'Enabled',
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/MenuView.js]: Shouldn't this be Is Enabled?

File: src/foam/u2/view/MenuView.js [hunk 2]
```diff
@@ -27,6 +27,27 @@ foam.CLASS({
...
+      help: 'Function to determine if button is enabled.',
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/MenuView.js]:
```suggestion
      documentation: 'Function to determine if button is enabled.',
```

File: src/foam/u2/view/MenuView.js (hunk 1 above)
Comment (noodlemoodle) [inline src/foam/u2/view/MenuView.js]:
```suggestion
      label: 'Is Enabled',
```

**Comment by jlhughes** [comment]
We have interface EnabledAware with property enabled. Let's be consistent. 
Realize you don't need the interface here, but for property naming,  'enabled' would be sufficient.

**Comment by noodlemoodle** [comment]
But this view is extending button so wouldn't isEnabled be more consistent? @jlhughes

File: src/foam/u2/view/MenuView.js [hunk 3]
```diff
@@ -27,6 +27,27 @@ foam.CLASS({
     {
       name: 'themeIcon',
       expression: function(menu) { return menu.themeIcon || ''; }
+    },
+    {
+      class: 'Function',
+      generateJava: false,
+      name: 'isEnabled',
+      label: 'Is Enabled',
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/MenuView.js]:
```suggestion
```
Already the default behaviour

**Comment by sarthak-marwaha** [comment]
LGTM, this has to be instance specific enabled/disabled as it enables the button for launching the menu based on the data of a given view, not the menu itself. Also isEnabled is the precedence was this kind of slotting in Actions so the name makes sense to me

**Review by jlhughes — APPROVED** [review]

**Pull request merged**